mlflow/utils/validation.py

```diff
@@ -70,6 +96,53 @@ def _validate_experiment_id(exp_id):
                               error_code=INVALID_PARAMETER_VALUE)
 
 
+def _validate_batch_limit(entity_name, limit, length):
+    if length > limit:
+        error_msg = ("A batch logging request can contain at most {limit} {name}. "
+                     "Got {count} {name}. Please split up {name} across multiple requests and try "
+                     "again.").format(name=entity_name, count=length, limit=limit)
+        raise MlflowException(error_msg, error_code=INVALID_PARAMETER_VALUE)
+
+
+def _validate_batch_log_limits(metrics, params, tags):
+    """Validate that the provided batched logging arguments are within expected limits."""
+    _validate_batch_limit(entity_name="metrics", limit=MAX_METRICS_PER_BATCH, length=len(metrics))
+    _validate_batch_limit(entity_name="params", limit=MAX_PARAMS_TAGS_PER_BATCH, length=len(params))
+    _validate_batch_limit(entity_name="tags", limit=MAX_PARAMS_TAGS_PER_BATCH, length=len(tags))
+    total_length = len(metrics) + len(params) + len(tags)
+    _validate_batch_limit(entity_name="metrics, params, and tags",
+                          limit=MAX_ENTITIES_PER_BATCH, length=total_length)
+
+
+def _validate_batch_log_data(metrics, params, tags):
+    for metric in metrics:
+        _validate_metric_name(metric.key)
+    for param in params:
+        _validate_param_name(param.key)
+        _validate_param_value(param.value)
+    for tag in tags:
+        _validate_tag_name(tag.key)
+        _validate_tag_value(tag.value)
+    # Verify upfront that the user isn't attempting to overwrite any params within their
+    # batched logging request
+    param_map = {}
+    for param in params:
+        if param.key not in param_map:
+            param_map[param.key] = param.value
+        elif param.key in param_map and param_map[param.key] != param.value:
+            raise MlflowException("Param %s had existing value %s, refusing to overwrite with "
```

Comment:
Can we reuse this same code in backend stores as well. So if there are any changes in future -- we change one place.